Content.Client/Temperature/Systems/EntityHeaterSystem.cs

```diff
@@ -1,5 +1,47 @@
+using Content.Shared.SS220.EntityEffects.Effects;
 using Content.Shared.Temperature.Systems;
+using Robust.Client.GameObjects;
 
 namespace Content.Client.Temperature.Systems;
 
-public sealed partial class EntityHeaterSystem : SharedEntityHeaterSystem;
+public sealed partial class EntityHeaterSystem : SharedEntityHeaterSystem
+{
+    //SS220-grill-update begin
+    [Dependency] private readonly SpriteSystem _spriteSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<GrillingVisualComponent, AfterAutoHandleStateEvent>(OnGrillingVisualAdd);
+        SubscribeLocalEvent<GrillingVisualComponent, ComponentShutdown>(OnGrillingVisualRemoved);
+    }
+
+    private void OnGrillingVisualAdd(Entity<GrillingVisualComponent> ent, ref AfterAutoHandleStateEvent args)
+    {
+        if (!TryComp<SpriteComponent>(ent, out var sprite))
+            return;
+
+        var layerData = new PrototypeLayerData()
+        {
+            RsiPath = ent.Comp.GrillingSprite?.RsiPath.ToString(),
+            State = ent.Comp.GrillingSprite?.RsiState,
+            MapKeys = [GrillingLayer]
+        };
+
+        if (_spriteSystem.TryGetLayer((ent.Owner, sprite), GrillingLayer, out var layer, true))
```

Comment:
```suggestion
        if (_spriteSystem.TryGetLayer((ent.Owner, sprite), GrillingLayer, out var layer, false))
```

Последний аргумент - это `logMissing`, выдаёт ошибку если не нашло лэйер. В нашем случае это не нужно